sal-service/src/main/java/org/ow2/proactive/sal/service/util/ByonUtils.java

```diff
@@ -244,18 +244,26 @@ public static String getBYONHostname(String nsName) {
     }
 
     /**
-     * Undeploy or remove the node source of a BYON node
-     * @param byonNode an object of class ByonNode to be undeployed or removed.
+     * Undeploy or remove the node source of BYON or Edge node
+     * @param id of ByonNode or EdgeNode to be undeployed or removed.
+     * @param nodeType the type of node source to be removed: Byon or Edge.
+     * @param systemArch the system architecture for Edge nodes.
      * @param preempt If true undeploy or remove node source immediately without waiting for nodes to be freed
      * @param remove If true completely remove the node source, if false only undeply the node source
      * @return  true if the resourceManagerGateway return no errors, false otherwise
      */
-    public static Boolean undeployByonNs(ByonNode byonNode, Boolean preempt, Boolean remove) {
-        assert byonNode != null : "A null value was passed for byonNode, A node source must have a BYON ID";
-        String nodeSourceName = "BYON_NS_" + byonNode.getId();
+    public static Boolean undeployNs(String id, String nodeType, String systemArch, Boolean preempt, Boolean remove) {
```

Comment:
Instead of making the method more complex to read, why not providing the nodeSourceName as parameter?

Reply:
done